src/api/item.ts

```diff
@@ -25,20 +58,24 @@ export async function createItem(
 export async function updateItem(
   subCategoryId: string,
   data: ItemType
-): Promise<{ item: ItemType } | undefined> {
+): Promise<{ subcategoryItem: ItemType } | undefined> {
   try {
-    const res = await fetch(
-      `${process.env.NEXT_PUBLIC_API_URL}subcategory-items/${subCategoryId}`,
-      {
-        method: 'PUT',
-        headers: {
-          Authorization: `Bearer ${process.env.NEXT_PUBLIC_TOKEN}`,
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify(data),
-      }
-    );
-    return res.json();
+    const token = cookies().get('token');
+    if (token && token.value) {
+      const res = await fetch(
+        `${process.env.NEXT_PUBLIC_API_URL}subcategory-items/${subCategoryId}`,
+        {
+          method: 'PUT',
+          headers: {
+            Authorization: `Bearer ${process.env.NEXT_PUBLIC_TOKEN}`,
```

Comment:
use **token.value** in all these calls  instead of **process.env.NEXT_PUBLIC_TOKEN**, otherwise another user will get these updates :D.